src/lib.rs

```diff
@@ -334,15 +334,22 @@ pub unsafe trait FromZeroes {
     where
         Self: Sized,
     {
+        let size = mem::size_of::<Self>()
+            .checked_mul(len)
+            .expect("mem::size_of::<Self>() * len overflows `usize`");
+        let align = mem::align_of::<Self>();
+        // On stable Rust versions <= 1.64.0, `Layout::from_size_align` has a
+        // bug in which sufficiently-large allocations (those which, when
+        // rounded up to the alignment, overflow `isize`) are not rejected,
+        // which can cause undefined behavior. See #64 for details.
+        //
+        // TODO(#67): Once our MSRV is > 1.64.0, remove this assertion.
+        let max_alloc = (usize::MAX / 2).saturating_sub(align);
```

Comment:
Nit: I think this would be clearer as `(isize::MAX as usize).saturating_sub(align)`

Reply:
Good call; done.